### PR TITLE
Feature/auth submodule

### PR DESCRIPTION
**Pull request by jaelim-ms**
* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Adding MSAL to App Center workspace as a git submodule and build it with Auth module.

### CODE REVIEW

File: Config/Tests iOS.xcconfig [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 
-OTHER_LDFLAGS = $(inherited) -framework UIKit -ObjC -framework CrashReporter -framework SafariServices -framework WebKit -framework AuthenticationServices -lz
+OTHER_LDFLAGS = $(inherited) -framework UIKit -framework SafariServices -framework WebKit -framework AuthenticationServices -lz
 
 LD_RUNPATH_SEARCH_PATHS = @executable_path/Frameworks @loader_path/Frameworks "$(SRCROOT)/../Vendor/iOS/OCMock" "$(SRCROOT)/../Vendor/iOS/OCHamcrest" "$(SRCROOT)/../Vendor/iOS/OHHTTPStubs"
 
 // OCMock and OCHamcrest have a bunch of warnings so we just disable this only for testing.
 GCC_TREAT_WARNINGS_AS_ERRORS = NO
+
+USER_HEADER_SEARCH_PATHS=$(inherited) $(SRCROOT)/MSAL/**
```
Comment (clpolet) [inline Config/Tests iOS.xcconfig]: Isn't that just for Identity? Can probably be moved to dedicated test config?

File: Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
-if [ -z $(otool -L "${INSTALL_DIR}/${FMK_NAME}" | grep 'libMSAL') ]; then
-  if [ -z "$MS_ARM64E_XCODE_PATH" ] || [ ! -d "$MS_ARM64E_XCODE_PATH" ]; then
-    echo "Use legacy Xcode and link MSAL via libtool."
-    libtool -static  "${INSTALL_DIR}/${FMK_NAME}" "${SRCROOT}/../Vendor/iOS/MSAL/MSAL.framework/MSAL" -o "${INSTALL_DIR}/${FMK_NAME}"
-  else
-    echo "Use arm64e Xcode and link MSAL via libtool."
-    env DEVELOPER_DIR="$MS_ARM64E_XCODE_PATH" /usr/bin/libtool -static  "${INSTALL_DIR}/${FMK_NAME}" "${SRCROOT}/../Vendor/iOS/MSAL/MSAL.framework/MSAL" -o "${INSTALL_DIR}/${FMK_NAME}"
-  fi
-fi
+source $(dirname "$0")/../build-ios-framework.sh AppCenterIdentity
```
Comment (clpolet) [inline Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh]: We don't need to merge MSAL lib anymore?

Comment (jaelim-ms) [reply Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh]: This is a project dependency. The binary will contain MSAL binary.

File: Config/Tests iOS.xcconfig (hunk 1 above)
Comment (jaelim-ms) [reply Config/Tests iOS.xcconfig]: Oops. Yeah.. I didn't realize that it is a common one. I have to revert one change that I made above and capture a work item for this one to separate.

**Pull request merged**